src/int.satyg

```diff
@@ -113,22 +129,39 @@ end = struct
     | Some(i) -> Some(i + 10)
     | None -> None))
 
-  let of-string ?:radix s =
-    let radix = match radix with
-    | None -> 10
-    | Some(radix) -> radix in
+
+  let to-string = arabic
+  let of-string-opt ?:radix-opt s =
+    let radix =
+      match radix-opt with
+      | None -> 10
+      | Some(i) -> i
+    in
     let cs = String.to-list s in
     match cs with
-    | [] -> panic `Int.of-string: invalid string`
-    | c :: cs2 ->
+    | [] -> None
+    | c::cs2 ->
       let (sign, cs) = if Eq.equal Char.eq c (Char.make `-`) then (-1, cs2) else (1, cs) in
-      let num = cs |> List.fold-left (fun acc c ->
-        (match char-to-int c with
-        | None -> panic (`Int.of-string: invalid string: `# ^ s)
-        | Some(i) ->
-        if i < radix then
-          acc * radix + i
-        else
-          panic (`Int.of-string: invalid string: `# ^ s))) 0 in
-      sign * num
+      let num-opt = cs |> List.fold-left (fun acc-opt c -> (
+        match acc-opt with
+        | None -> Option.none
+        | Some(acc) -> (
+          match char-to-int c with
+          | None -> Option.none
+          | Some(i) ->
+          if i < radix then
+            Option.some (acc * radix + i)
+          else
+            Option.none
+        ))) (Option.some 0) in
+      match num-opt with
+      | None -> Option.none
+      | Some(num) -> Option.some (sign * num)
+
+  let of-string ?:radix-opt s =
+    match radix-opt with
+    | None -> of-string-opt s |> Option.expect `Int.of-string: nanka okasii`
```

Comment:
I said `nanka okasii` just for example 🤣 
Can you change it to say `Int.of-string: invalid string: `# ^ s

Reply:
Apparently, I misunderstood.😯